src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -264,21 +264,7 @@ class TypeApplications(val self: Type) extends AnyVal {
         self.cls.typeParams
       case self: TypeRef =>
         val tsym = self.symbol
-        if (tsym.isClass) tsym.typeParams
-        else tsym.infoOrCompleter match {
-          case completer: TypeParamsCompleter =>
```

Comment:
So why isn't this needed anymore? Does this mean that the `TypeParamsCompleter` class can be deleted?